### DISCUSSION TITLE
esp32-A1S (Audio Kit) vs Normal esp32 with separate modules

### DISCUSSION THREAD

**Discussion by jazzbiddy** [marcel-licence/esp32_midi_sampler#40] [General]
This is a general question, but what are the benefits of using the esp32 audiokit instead of just using a normal esp32 with an sd card module and i2s headphone or speaker module?  I'm asking mostly because of the size that the esp32 audiokit board seems to be bigger than the individual parts and will take up more room in my enclosure than I'd like. I'm doubting it's value. the main goal of my project is to make a drum trigger like sample player - where incoming midi notes trigger sounds assigned to them.  

Is the AC101 encoder such a big deal?  Is it much better sound quality than normal I2S? Does it allow for more effects applied to the wav files?  Aside from everything being in on board - are there any performance improvements or built in cache or storage improvements with the audio kit? 

Thanks for any thoughts.

**Comment by souravlal** [comment]
I think you can buy ESP32 A1S Module instead. Which will have both ESP32 processor and AC101 audio codec equipped inside.

https://www.electronicscomp.com/ai-thinker-esp32-a1s-wifi-bt-audio-development-board

![51OOldWGNlL _SY450_](https://user-images.githubusercontent.com/53053463/127443522-da6acb61-3a4a-4115-ba5e-01cc7703a678.jpg)

**Reply by marcel-licence** [reply]
It could be smaller =) The advantage is that SD Card and audio connectors and also an amplifier is also on board. The serial interface for programming and logging is also placed on the board. I also used the line in and also the microphones for testing. With breakout boards the result would be much bigger (when using all features). Using an SD card with SD MMC interface using a breakout failed in my tests.
The processing power would be the same with an ESP32 (you may need PSRAM on board). I2S is a digital interface so you could use a lot of different DACs. The AC101 has also an additional I2C connection for configuration. You can for example mix line-in to your output. There are a lot of options. I will also try to build up another hardware soon.

**Reply by jazzbiddy** [reply]
These are very good points.  

@souravlal I think it's way too much work to figure out how to build a pcb starting with just the small module. I'd prefer to be making music ;) 

@marcel-licence I'm still experimenting with all of this, but did see you had some videos where you were using the normal esp32.  You're probably right that, in the end, it might take up just as much room or not give the better results - but it's the results part that I'm wondering about.  So - in your opinion - you would use the esp32-A1S for future "sampler" projects instead of a normal esp32 with an external sdcard and external I2S amp with headphone/speaker?

**Reply by souravlal** [reply]
ESP32 A1S circuit diagram 👇

[esp32-audio-kit_v2.2_sch.pdf](https://github.com/marcel-licence/esp32_midi_sampler/files/6904913/esp32-audio-kit_v2.2_sch.pdf)

**Reply by marcel-licence** [reply]
At the moment I am using the ESP32 Audio Kit very often because most of the components I am using are available. But I am also creating projects where you can use smaller hardware (without SD, without audio input). You can interface a lot components also with a smaller ESP: https://youtu.be/juwm6o3j8XA
It looks like the Audio Kit may be discontinued or replaced by a new version with another codec. Using separate components would be easier to use other components.
